pandas/core/algorithms.py

```diff
@@ -395,6 +395,14 @@ def isin(comps, values):
                         " to isin(), you passed a [{values_type}]"
                         .format(values_type=type(values).__name__))
 
+    # GH 25507
+    # if `values` is a set, directly use it instead of hashing a list
+    if isinstance(values, set):
```

Comment:
i think we have an is_set_like (otherwise pls create one)

Reply:
I didn't find an existing one so I have added it.